control-plane/rest/service/src/v0/snapshots.rs

```diff
@@ -39,35 +37,117 @@ impl apis::actix_server::Snapshots for RestApi {
+        Path(volume_id): Path<Uuid>,
+    ) -> Result<models::VolumeSnapshots, RestError<RestJsonError>> {
+
+        let snaps = client().get_snapshots(Filter::Volume(volume_id.into()), true, None).await?;
+        Ok(models::VolumeSnapshots {
+            entries: snaps.entries.into_iter().map(|e| e.into()).collect(),
+        })
     }
 
     async fn get_volumes_snapshot(
-        Path(_snapshot_id): Path<Uuid>,
-    ) -> Result<VolumeSnapshot, RestError<RestJsonError>> {
-        Err(ReplyError::unimplemented("Snapshot service is not implemented".to_string()).into())
+        Path(snapshot_id): Path<Uuid>,
+    ) -> Result<RestSnap, RestError<RestJsonError>> {
+        let snaps = client().get_snapshots(Filter::Snapshot(snapshot_id.into()), true, None).await?;
+        let snap = if let Some(s) = snaps.entries().first() {
+            s
+        } else {
+            return Err(ReplyError::not_found(
+                ResourceKind::VolumeSnapshot,
+                "Snapshot not found".to_string(),
+                snapshot_id.to_string(),
+            ).into())
+        };
+
+        Ok(RestSnap {
+            definition: models::VolumeSnapshotDefinition::new_all(
+                models::VolumeSnapshotMetadata::new_all(
+                    snap.meta().timestamp().map(|t| t.to_string()),
+                    snap.meta().txn_id(),
+                    std::collections::HashMap::new(),
+                ),
+                models::VolumeSnapshotSpec::new_all(snap.spec().snap_id(), snap.spec().source_id()),
+            ),
+            state: models::VolumeSnapshotState::new_all(
+                snap.state().uuid(),
+                snap.state().size(),
+                snap.state().source_id(),
+                snap.state()
+                    .timestamp()
+                    .map(|t| t.to_string())
+                    .unwrap_or_default(),
+                snap.state().clone_ready(),
+                Vec::<models::ReplicaSnapshotState>::new(),
+            ),
+        })
     }
 
     async fn get_volumes_snapshots(
-        Query((_snapshot_id, _volume_id)): Query<(Option<Uuid>, Option<Uuid>)>,
-    ) -> Result<Vec<VolumeSnapshot>, RestError<RestJsonError>> {
-        Err(ReplyError::unimplemented("Snapshot service is not implemented".to_string()).into())
+        Query((snapshot_id, volume_id)): Query<(Option<Uuid>, Option<Uuid>)>,
+    ) -> Result<models::VolumeSnapshots, RestError<RestJsonError>> {
+        let filter: Filter;
+
+        if snapshot_id.is_some() && volume_id.is_some() {
+            filter = Filter::VolumeSnapshot(volume_id.unwrap().into(), snapshot_id.unwrap().into());
+        } else if snapshot_id.is_some() && volume_id.is_none() {
+            filter = Filter::Snapshot(snapshot_id.unwrap().into());
+        } else if snapshot_id.is_none() && volume_id.is_some() {
+            filter = Filter::Volume(volume_id.unwrap().into());
+        } else {
+            filter = Filter::None;
+        }
```

Comment:
```suggestion
       let filter = match (snapshot_id, volume_id) {
    (Some(snapshot), Some(volume)) => Filter::VolumeSnapshot(volume.into(), snapshot.into()),
    (Some(snapshot), None) => Filter::Snapshot(snapshot.into()),
    (None, Some(volume)) => Filter::Volume(volume.into()),
    _ => Filter::None,
};
```

Nit:- A bit easy to read.